### PR TITLE
SIDE-988 - Update Go docs for new loops

### PR DESCRIPTION
**Pull request by dmorrison-olive**
https://crosschx.atlassian.net/browse/SIDE-988

### CODE REVIEW

File: ldk/go/readme.md [hunk 1]
```diff
@@ -79,13 +75,13 @@ It can be useful to tail the log as you develop to see any errors, warnings, or
...
 ```sh
-tail -f /var/log/Sidekick/Sidekick-2020.08.12-1dcc37a.log
+tail -F /var/log/Sidekick/*.log
 ```
 
 ##### Mac
 
 ```sh
-tail -f ~/Library/Logs/Sidekick/Sidekick-2020.08.12-1dcc37a.log
+tail -F ~/Library/Logs/Sidekick/*.log
```
Comment (richseviora) [inline ldk/go/readme.md:84]: I think this has changed to `~/Library/Logs/Olive\ Helps/`

**Comment by dmorrison-olive** [comment]
Might be helpful to view these as rendered Markdown on this branch: https://github.com/open-olive/loop-development-kit/tree/SIDE-988-update-go-docs/ldk/go

**Comment by dmorrison-olive** [comment]
I haven't touched `docs/dataFlow.drawio.svg` yet. I just noticed where I can load and update it on draw.io, so I'll try that.

File: ldk/go/docs/loops.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		return err
+	}
+}
+```
+
+Loops do not need to emit whispers in a 1:1 relationship with events. Loops may not use events at all. Loops may only use some events. Loops may keep a history of events and only emit whispers when several conditions are met.
+
+## Lifecycle
+
+1. Sidekick executes plugin process.
```
Comment (richseviora) [inline ldk/go/docs/loops.md:46]: OK - so... there's a bunch of renaming that needs to be done here. I think that'd be covered within SIDE-988 right? 😄 

Soo, could you go through and replace all references to "Sidekick" with "Olive Helps", excluding code references? I'll need to rename the `Sidekick` struct at some point, but anything in docs should be renamed.

File: ldk/go/readme.md [hunk 3]
```diff
@@ -79,13 +75,13 @@ It can be useful to tail the log as you develop to see any errors, warnings, or
 ##### Linux
 
 ```sh
-tail -f /var/log/Sidekick/Sidekick-2020.08.12-1dcc37a.log
+tail -F /var/log/Sidekick/*.log
```
Comment (richseviora) [inline ldk/go/readme.md:78]: This has also changed

File: ldk/go/readme.md [hunk 4]
```diff
@@ -96,7 +92,7 @@ From `Powershell` (not a cmd shell):
 Get-Content $env:AppData\Sidekick\Logs\Sidekick-2020.08.12-1dcc37a.log –Wait
```
Comment (richseviora) [inline ldk/go/readme.md:92]: Also here

**Review by richseviora — CHANGES_REQUESTED** [review]
Some changes to naming

**Comment by iangallOlive** [comment]
@dmorrison-olive If you use VS Code, there's an [extension](https://marketplace.visualstudio.com/items?itemName=hediet.vscode-drawio) that you can use right in the IDE

**Comment by dmorrison-olive** [comment]
> @dmorrison-olive If you use VS Code, there's an [extension](https://marketplace.visualstudio.com/items?itemName=hediet.vscode-drawio) that you can use right in the IDE

Thanks for the tip! I'll check it out.

File: ldk/go/readme.md (hunk 1 above)
Comment (dmorrison-olive) [reply ldk/go/readme.md]: Gotcha - thanks for the heads up! I just changed it.

File: ldk/go/docs/loops.md (hunk 2 above)
Comment (dmorrison-olive) [reply ldk/go/docs/loops.md:46]: Sounds good to me - I can rename from "Sidekick" to "Olive Helps".

Just to be clear - I **won't* change the code samples in the docs yet and will wait for that to be done when the references throughout the LDK are changed from `Sidekick` to e.g. `OliveHelps`, right?

**Review by BillyMurrayOlive — APPROVED** [review]

File: ldk/go/readme.md (hunk 3 above)
Comment (dmorrison-olive) [reply ldk/go/readme.md]: I updated this. 👍

File: ldk/go/readme.md (hunk 4 above)
Comment (dmorrison-olive) [reply ldk/go/readme.md]: This was updated as well.

**Comment by dmorrison-olive** [comment]
I updated places _in the documentation_ to say "Olive Helps" instead of "Sidekick", and I left code samples as `Sidekick` (until that's updated in code).

I saw a bunch of places that say "plugin", and I thought about changing them to say "loop", but it got confusing to me. I left that alone, but maybe we should change it. ¯\_(ツ)_/¯

**Comment by dmorrison** [comment]
I updated the data-flow diagram. Please check that to make sure it aligns with reality. :)

**Review by richseviora — APPROVED** [review]

**Pull request merged**